### PR TITLE
fix(dashboard): fix selection bug 

### PR DESCRIPTION
**Pull request by jmbuss**
## Overview
This PR addresses 3 bugs.
1. The selection box not disappearing on completing the selection action. This was because the gesture was not reactive.
2. Clicking on an element selects all elements that overlap that pixel (not the top most element). To fix this we can sort by the z index and select the highest one in the selection.
3. Paste at position with a selection pastes all the widgets at that position (not the selection at that position). To fix this we can offset the copy group by the difference of paste position.

## Tests
[Include a link to the passing GitHub action running the test suite here.]

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

**Comment by diehbria** [comment]
it seems like theres some missing tests since there were no tests failing prior for the previous behavior but none added

**Review by diehbria — CHANGES_REQUESTED** [review]
add relevant tests

File: packages/dashboard/src/components/iot-dashboard/iot-dashboard-internal.tsx [hunk 1]
```diff
@@ -125,12 +127,19 @@ export class IotDashboardInternal {
...
   };
+
   onSendToBack = () => {
     this.sendToBack({
       widgets: this.dashboardConfiguration.widgets.filter(({ id }) => this.selectedWidgetIds.includes(id)),
     });
   };
 
+  onClearSelection = () => {
+    this.selectWidgets({
```
Comment (boweihan) [inline packages/dashboard/src/components/iot-dashboard/iot-dashboard-internal.tsx:138]: Are we planning to have integration tests for this type of functionality?

Comment (jmbuss) [reply packages/dashboard/src/components/iot-dashboard/iot-dashboard-internal.tsx:138]: ahh good catch, I missed those tests. will add.

**Review by diehbria — APPROVED** [review]
